llvm/lib/Target/M68k/M68kISelDAGToDAG.cpp

```diff
@@ -708,6 +708,20 @@ bool M68kDAGToDAGISel::SelectARIPD(SDNode *Parent, SDValue N, SDValue &Base) {
   return false;
 }
 
+static bool allowARIDWithDisp(SDNode *Parent) {
```

Comment:
will the compiler prompt unused function warning when building LLVM without assertion?

Reply:
style guide was unclear about what do in this case, so I added `[[maybe_unused]]` here